tests_recording/test_api.py

```diff
@@ -20,41 +20,109 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 
-import unittest
+import os
 from subprocess import check_output
-
-import rebasehelper
 from flexmock import flexmock
 from rebasehelper.exceptions import RebaseHelperError
 from requre.cassette import DataTypes
+from requre.online_replacing import (
+    apply_decorator_to_all_methods,
+    record_requests_for_all_methods,
+    replace_module_match,
+)
+from requre.helpers.tempfile import TempFile
+from requre.helpers.simple_object import Simple
+from requre.helpers.files import StoreFiles
+from requre.helpers.git.pushinfo import PushInfoStorageList
+from requre.helpers.git.fetchinfo import FetchInfoStorageList
+from requre.helpers.git.repo import Repo
 
 from packit.api import PackitAPI
-from tests_recording.testbase import PackitUnittestOgr
+from tests_recording.testbase import PackitTest
 
 
-@unittest.skip("Not working yet")
-class ProposeUpdate(PackitUnittestOgr):
-    def setUp(self):
-        if (
-            hasattr(rebasehelper, "VERSION")
-            and int(rebasehelper.VERSION.split(".")[1]) >= 19
-        ):
-            self.cassette.data_miner.key = "rebase-helper>=0.19"
-        else:
-            self.cassette.data_miner.key = "rebase-helper<0.19"
-        self.cassette.data_miner.data_type = DataTypes.Dict
+@record_requests_for_all_methods()
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="packit.utils.run_command_remote", decorate=Simple.decorator_plain()
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="packit.fedpkg.FedPKG.clone",
+        decorate=StoreFiles.where_arg_references(
+            key_position_params_dict={"target_path": 2}
+        ),
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="git.repo.base.Repo.clone_from",
+        decorate=StoreFiles.where_arg_references(
+            key_position_params_dict={"to_path": 2},
+            return_decorator=Repo.decorator_plain,
+        ),
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="git.remote.Remote.push", decorate=PushInfoStorageList.decorator_plain()
+    )
+)
+@apply_decorator_to_all_methods(
+    replace_module_match(
+        what="git.remote.Remote.fetch", decorate=FetchInfoStorageList.decorator_plain()
+    )
+)
```

Comment:
Looks like can create a shortcut for git.

And maybe also for `mkdtemp+mktemp`.